src/crypto/keccak.c

```diff
@@ -212,3 +212,51 @@ void keccak_finish(KECCAK_CTX * ctx, uint8_t *md){
         memcpy_swap64le(md, ctx->hash, KECCAK_DIGESTSIZE / sizeof(uint64_t));
     }
 }
+
+void hmac_keccak_init(hmac_keccak_state *S, const uint8_t *_key, uint64_t keylen) {
```

Comment:
Is this standard HMAC ? If so, it should probably in a hmac.[hc] file so it be used nicely by other stuff in the future.

Reply:
Correct, it is standard HMAC RFC 2104. I've ported it from `src/crypto/blake256.c`checking also against the specification.

I will move it to another file and add comments it is RFC 2104 HMAC.